builtin/logical/aws/path_roles.go

```diff
@@ -261,6 +268,16 @@ func (b *backend) pathRolesWrite(ctx context.Context, req *logical.Request, d *f
 		roleEntry.ProhibitFlexibleCredPath = false
 	}
 
+	if _, ok := d.GetOk("user_path"); ok {
+		if legacyRole != "" {
+			return logical.ErrorResponse("cannot supply deprecated role or policy parameters with user_path"), nil
+		}
+		if !strutil.StrListContains(roleEntry.CredentialTypes, iamUserCred) {
+			return logical.ErrorResponse(fmt.Sprintf("user_path parameter only valid for %s credential type", iamUserCred)), nil
+		}
```

Comment:
Thoughts on putting some rudimentary validations on the value supplied here? I'm usually not a fan of adding too many validations on AWS's requirements, and just letting AWS validate the values themselves. However, in the case, I think the [AWS requirements](https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateUser.html) are pretty generic and unlikely to change, and it would be a poor user experience to let somebody configure Vault with an invalid path and then only later discover the role were unusable due to the invalid path.

Reply:
Makes sense, I will add them :)